pythonforandroid/recipe.py

```diff
@@ -1037,17 +1011,147 @@ def get_recipe_env(self, arch, with_flags_in_cc=True):
 
 
 class TargetPythonRecipe(Recipe):
-    '''Class for target python recipes. Sets ctx.python_recipe to point to
-    itself, so as to know later what kind of Python was built or used.'''
+    '''
+    Class for target python recipes. Sets ctx.python_recipe to point to itself,
+    so as to know later what kind of Python was built or used.
+
+    This base class is used for our main python recipes (python2 and python3)
+    which shares most of the build process.
+
+    .. note:: The recipe for python3Crystax also inherits from this base class,
+        but overwrites most of the code used here because has some different
+        build process.
+
+    .. versionchanged:: 0.6.0
+
+        - Adds the python's build process, which has been taken from the
+          python3's recipe by `inclement` and refactored here, split
+          into several methods: :meth:`build_arch` and :meth:`get_recipe_env`.
+        - Adds the attribute :attr:`configure_args`, which has been moved from
+          the original python3's recipe method :meth:`build_arch` into a static
+          class variable.
+        - Changes the return value for methods: :meth:`include_root` and
+          :meth:`link_root`.
+        - Adds some static class variables used to create the python bundle and
+          modifies the method :meth:`create_python_bundle`, refactored from the
+          python3's recipe by inclement. The added variables, also taken from
+          the mentioned recipe, are: :attr:`stdlib_dir_blacklist`,
+          :attr:`stdlib_filen_blacklist`, :attr:`site_packages_dir_blacklist`
+          and :attr:`site_packages_filen_blacklist`.
+    '''
 
     from_crystax = False
     '''True if the python is used from CrystaX, False otherwise (i.e. if
     it is built by p4a).'''
 
+    configure_args = ()
+    '''The configure arguments needed to build the python recipe. Those are
+    used in method :meth:`build_arch` (if not overwritten like python3crystax's
+    recipe does).
+    
+    .. note:: This variable should be properly set in subclass.
+    '''
+
+    stdlib_dir_blacklist = {
+        '__pycache__',
+        'test',
+        'tests',
+        'lib2to3',
+        'ensurepip',
+        'idlelib',
+        'tkinter',
+    }
+    '''The directories that we want to omit for our python bundle'''
+
+    stdlib_filen_blacklist = [
```

Comment:
Why not also having it as a set? I actually liked the idea

Reply:
Because I suspect that @inclement did it on purpose, this way the data types for the "file extensions blacklist" match, and notice that one of this lists is empty and, as far as I know we cannot initialize an empty set with {}, we should use set(), I think that is more elegant the way it is  :wink: